src/deviceApiCalls/transports/android.transport.js

```diff
@@ -0,0 +1,117 @@
+import {DeviceApiTransport} from '../../../packages/device-api'
+import {
+    GetAutofillDataCall,
+    GetAvailableInputTypesCall,
+    GetRuntimeConfigurationCall,
+    StoreFormDataCall
+} from '../__generated__/deviceApiCalls'
+
+export class AndroidTransport extends DeviceApiTransport {
+    /** @type {GlobalConfig} */
+    config
+
+    /** @param {GlobalConfig} globalConfig */
+    constructor (globalConfig) {
+        super()
+        this.config = globalConfig
+    }
+    /**
+     * @param {import("../../../packages/device-api").DeviceApiCall} deviceApiCall
+     * @returns {Promise<any>}
+     */
+    async send (deviceApiCall) {
+        if (deviceApiCall instanceof GetRuntimeConfigurationCall) {
+            return androidSpecificRuntimeConfiguration(this.config)
+        }
+
+        if (deviceApiCall instanceof GetAvailableInputTypesCall) {
+            return androidSpecificAvailableInputTypes(this.config)
+        }
+
+        if (deviceApiCall instanceof GetAutofillDataCall) {
+            window.BrowserAutofill.getAutofillData(JSON.stringify(deviceApiCall.params))
+            return waitForResponse(deviceApiCall.id, this.config)
```

Comment:
Just being paranoid. Is it possible that the native layer responds before we have the chance to register the listener? I don't think so, but wanted to double check with you.

Reply:
Not from what I've experienced so far on Android + Windows - once merged and Android does more testing on their end we can always circle back here. there's no risk to the existing android application because we havn't changed how the email protection calls works 🙏🏻